core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlResponseParserUtils.java

```diff
@@ -49,7 +53,7 @@ public static XmlElement parse(SdkPojo sdkPojo, SdkHttpFullResponse response) {
             // In some cases the responseContent is present but empty, so when we are not expecting a body we should
             // not attempt to parse it even if the body appears to be present.
             if ((!response.isSuccessful() || hasPayloadMembers(sdkPojo)) && responseContent.isPresent() &&
-                !contentLengthZero(response)) {
+                    !contentLengthZero(response) && !getBlobTypePayloadMemberToUnmarshal(sdkPojo).isPresent()) {
```

Comment:
This is long with a lot of nesting. Can we extract it and apply some names to the intermediate values?

Reply:
removed context.response().content().isPresent check as we already do in [LineNumber 83](https://github.com/aws/aws-sdk-java-v2/blob/9cb352e43d72eca4bb64254577a1ae039b9affd8/core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java#L83-L86).
Replaced with local variable